packages/cspell-trie-lib/src/lib/suggest.ts

```diff
@@ -309,5 +309,5 @@ export function suggestionCollector(
  * note: suppress js/incomplete-sanitization warning about SQL injection.
  */
 function regexQuote(text: string): string {
-    return text.replace(/[\[\]\-+(){},|*.]/g, '\\$1'); // lgtm[js/incomplete-sanitization]
+    return text.replace(/[\[\]\-+(){},|*.]/g, '\\$1'); // lgtm
```

Comment:
I think it's asking for the backslash to be escaped. I think it sees the `\-` as escaping the `-`. alternately, you could move the `-` as the last character in the array
```suggestion
    return text.replace(/[\[\]\-+(){},|*.\\]/g, '\\$1'); // lgtm
```
Didn't actually test this though

Reply:
I'll try that, thank you. Adding `\\` also makes sense.